src/Type/ValidatorResolverDynamicReturnTypeExtension.php

```diff
@@ -0,0 +1,46 @@
+<?php declare(strict_types = 1);
+
+namespace SaschaEgerer\PhpstanTypo3\Type;
+
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Name;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use PHPStan\Type\DynamicMethodReturnTypeExtension;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\Type;
+use TYPO3\CMS\Extbase\Validation\ValidatorResolver;
+
+class ValidatorResolverDynamicReturnTypeExtension implements DynamicMethodReturnTypeExtension
+{
+
+	public function getClass(): string
+	{
+		return ValidatorResolver::class;
+	}
+
+	public function isMethodSupported(
+		MethodReflection $methodReflection
+	): bool
+	{
+		return $methodReflection->getName() === 'createValidator';
+	}
+
+	public function getTypeFromMethodCall(
+		MethodReflection $methodReflection,
+		MethodCall $methodCall,
+		Scope $scope
+	): Type
+	{
+		$arg = $methodCall->args[0]->value;
+		if (!($arg instanceof \PhpParser\Node\Expr\ClassConstFetch)) {
+			return ParametersAcceptorSelector::selectSingle($methodReflection->getVariants())->getReturnType();
+		}
+		/** @var Name $class */
+		$class = $arg->class;
+
+		return new ObjectType((string) $class);
```

Comment:
This should be `return TypeCombinator::addNull(new ObjectType((string) $class));`, right?
As far as i can see `ValidatorResolver ::createValidator()` may return null if the requested validator does not exist

Reply:
Your totally right. I'm new to phpstan, so didn't notice that. Have pushed an amended version of the commit.